src/main/java/com/mongodb/hibernate/internal/translate/AbstractMqlTranslator.java

```diff
@@ -143,12 +160,14 @@ abstract class AbstractMqlTranslator<T extends JdbcOperation> implements SqlAstT
 
     private final List<JdbcParameterBinder> parameterBinders = new ArrayList<>();
 
+    private final Set<String> affectedTableNames = new HashSet<>();
+
     AbstractMqlTranslator(SessionFactoryImplementor sessionFactory) {
         this.sessionFactory = sessionFactory;
-        assertNotNull(sessionFactory
-                .getServiceRegistry()
-                .requireService(StandardServiceRegistryScopedState.class)
-                .getConfiguration());
+        /*assertNotNull(sessionFactory
+        .getServiceRegistry()
+        .requireService(StandardServiceRegistryScopedState.class)
+        .getConfiguration());*/
```

Comment:
The above code is not used currently and mocking them requires lots of efforts. For now the block is commented out and we can uncomment it in the future when it is required.

Reply:
Generally I think we should prefer Jira tickets rather than commented out code, which has a way of living forever and confusing readers.

Reply:
> The above code is not used currently

It is used - it's an assertion that makes sure `MongoConfiguration` is accessible where it may be needed.

Mocking service registry is not so "lots of efforts", that it warrants removing that assertion. However, we may not need to do this at all, because we can trigger auto-flush and `AbstractMqlTranslator.getAffectedTableNames` (it's even simpler than what I proposed), so we may be able to create an integration test for that (I am experimenting).

Reply:
I was too optimistic. The original idea failed (see https://github.com/mongodb/mongo-hibernate/pull/69#discussion_r2004584489), but I thought the suggestions from https://docs.jboss.org/hibernate/orm/6.6/javadocs/org/hibernate/Session.html#get(java.lang.Class,java.lang.Object) would help, but they didn't, which seems obvious to me now.

Given that we are stuck with the new `SelectStatementMqlTranslatorTests.testAffectedTableNames`, at least for now, let's reinstate the assertion and get rid of the "visible for testing" `JdbcOperationQuerySelect.getAggregateCommand` method. The proposed change is in https://github.com/stIncMale/mongo-hibernate/commit/6522e2608762d217aa6fd5c5396ab8eb9c504ac6.

Reply:
adopted. thanks.

Reply:
@NathanQingyangXu, you started this thread, so I am not resolving it. But it can be resolved, so please feel free.

Reply:
thanks for reminding. I resolved.